### PR TITLE
Down mattress Typo fix and makes it interchangeable component as mattress in construction

### PR DESCRIPTION
**Pull request by Maleclypse**
#### Summary
`SUMMARY: Content "Down Mattress typo and bed construction"`
#### Purpose of change
Makes down mattresses interchangeable for bed construction purposes and fixes typo in naming convention

#### Describe the solution
I noticed that in an expiremental where down mattresses can be built they couldn't be used to make real beds.  I believe this will fix this. 

#### Describe alternatives you've considered
None, this was an original goal of creating down mattresses but I didn't know enough at the time to make it happen.

### CODE REVIEW

File: data/json/items/resources/misc.json [hunk 1]
```diff
@@ -37,33 +37,17 @@
     "id": "down_mattress",
     "type": "GENERIC",
     "category": "spare_parts",
-    "name": "down_mattress",
-    "name_plural": "down_mattresses",
+
```
Comment (Qrox) [inline data/json/items/resources/misc.json]:
```suggestion
```

File: data/json/items/resources/misc.json [hunk 2]
```diff
@@ -37,33 +37,17 @@
...
-    "volume": "80 ml",
-    "price": 500,
-    "container": "bag_plastic",
-    "material": [ "cotton" ],
-    "symbol": ",",
-    "color": "dark_gray",
-    "count": 50,
-    "ammo_type": "components",
-    "effects": [ "COOKOFF" ]
+
```
Comment (Qrox) [inline data/json/items/resources/misc.json]:
```suggestion
```

File: data/json/items/resources/misc.json [hunk 3]
```diff
@@ -37,33 +37,17 @@
...
     "weight": 20000,
     "volume": "300000 ml",
     "price": 1000,
     "material": [ "cotton" ],
     "symbol": "0",
     "color": "white",
-    "use_action": { "type": "deploy_furn", "furn_type": "f_down_mattress" }
-  },
-  {
-    "id": "fuse",
```
Comment (Qrox) [inline data/json/items/resources/misc.json]: Why is this removed?

Comment (Maleclypse) [reply data/json/items/resources/misc.json]: I was getting the fuse item as an error that needed to be resolved so when I looked in the experimental version I've been playing it looks like fuse had already been removed.  But I could just be looking at too many different versions and getting timeline confused.

**Comment by Maleclypse** [comment]
@davidpwbrown I thought I was implementing the either/or component correctly but the CI build looks like it disagrees? If you have a second sometime do you mind checking? Thanks and no worries if not!

**Comment by Maleclypse** [comment]
another thought is would it be better to use the `copy from` mattress in the down mattress item?

**Comment by ZhilkinSerg** [comment]
Please test your changes in-game. Game won't load due to invalid json structure.

**Comment by Maleclypse** [comment]
> Please test your changes in-game. Game won't load due to invalid json structure.

Can do.  I thought the easiest place to see the failure details was via github reports.  Sorry if it's causing some issues that I've been doing it here rather than loading into a set aside, experimental locally.  I'll change my practice.

**Comment by ZhilkinSerg** [comment]
> 
> 
> > Please test your changes in-game. Game won't load due to invalid json structure.
> 
> Can do. I thought the easiest place to see the failure details was via github reports. Sorry if it's causing some issues that I've been doing it here rather than loading into a set aside, experimental locally. I'll change my practice.

You need to test your own changes - at least make sure game compiles (if code was changed), starts and loads without errors/crashes. There is no sense to wait until CI or mergetester will find out that it is failing.

File: data/json/items/resources/misc.json (hunk 3 above)
Comment (AMurkin) [reply data/json/items/resources/misc.json]: You should update master branch from upstream and rebase this branch.

**Comment by Maleclypse** [comment]
> > > Please test your changes in-game. Game won't load due to invalid json structure.
> > 
> > 
> > Can do. I thought the easiest place to see the failure details was via github reports. Sorry if it's causing some issues that I've been doing it here rather than loading into a set aside, experimental locally. I'll change my practice.
> 
> You need to test your own changes - at least make sure game compiles (if code was changed), starts and loads without errors/crashes. There is no sense to wait until CI or mergetester will find out that it is failing.

Thanks for this.  I've learned from this that I have a copy/paste issue where jsonstyle is carried over from atom to a playable version but if I then take changes from that back into Atom it needs to be relinted first.  Which is probably due to me using notepad when I'm editing in a playable version.

File: data/json/items/resources/misc.json (hunk 3 above)
Comment (Maleclypse) [reply data/json/items/resources/misc.json]: If this last change fails I think I'm going to close this PR and do what you are saying.  Or can I rebase this branch with the PR open? Will that cause more problems?

Comment (AMurkin) [reply data/json/items/resources/misc.json]: You don't need to close PR to rebase. And you may check if your local branch is ok after rebasing before pushing changes to Github.
